internal/provider/data_source_gitlab_project_issues.go

```diff
@@ -0,0 +1,379 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mitchellh/hashstructure"
+	"github.com/xanzy/go-gitlab"
+)
+
+var validIssueOrderByValues = []string{
+	"created_at",
+	"updated_at",
+	"priority",
+	"due_date",
+	"relative_position",
+	"label_priority",
+	"milestone_due",
+	"popularity",
+	"weight",
+}
+
+var validIssueScopeValues = []string{"created_by_me", "assigned_to_me", "all"}
```

Comment:
For this and other resources, can you move your `valid*` variables into the `register*` function itself, to keep the global scope clean? This is mostly a readability concern for me.